numpy/random/common.pxd

```diff
@@ -0,0 +1,101 @@
+#cython: language_level=3
+
+from libc.stdint cimport (uint8_t, uint16_t, uint32_t, uint64_t,
+                          int8_t, int16_t, int32_t, int64_t, intptr_t,
+                          uintptr_t)
+from libc.math cimport sqrt
+
+from .distributions cimport bitgen_t 
+import numpy as np
+cimport numpy as np
+
+cdef double POISSON_LAM_MAX
+cdef uint64_t MAXSIZE
+
+cdef enum ConstraintType:
+    CONS_NONE
+    CONS_NON_NEGATIVE
+    CONS_POSITIVE
+    CONS_POSITIVE_NOT_NAN
+    CONS_BOUNDED_0_1
+    CONS_BOUNDED_0_1_NOTNAN
+    CONS_BOUNDED_GT_0_1
+    CONS_GT_1
+    CONS_GTE_1
+    CONS_POISSON
+
+ctypedef ConstraintType constraint_type
+
+cdef object benchmark(bitgen_t *bitgen, object lock, Py_ssize_t cnt, object method)
+cdef object random_raw(bitgen_t *bitgen, object lock, object size, object output)
+cdef object prepare_cffi(bitgen_t *bitgen)
+cdef object prepare_ctypes(bitgen_t *bitgen)
+cdef int check_constraint(double val, object name, constraint_type cons) except -1
+cdef int check_array_constraint(np.ndarray val, object name, constraint_type cons) except -1
+
+cdef extern from "src/aligned_malloc/aligned_malloc.h":
+    cdef void *PyArray_realloc_aligned(void *p, size_t n)
```

Comment:
NumPY should consider adding these to the official API maybe.  @sturlamolden wrote these.